zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -325,13 +337,38 @@ where
             // Convert into TransactionTemplates
             let mempool_txs = mempool_txs.iter().map(Into::into).collect();
 
-            let empty_string = String::from("");
+            // Calling state chainfo to additional data
+            let request = ReadRequest::ChainInfo();
+            let response = state
+                .ready()
+                .and_then(|service| service.call(request))
+                .await
+                .map_err(|error| Error {
+                    code: ErrorCode::ServerError(0),
+                    message: error.to_string(),
+                    data: None,
+                })?;
+
+            let chain_info = match response {
+                ReadResponse::ChainInfo(chain_info) => chain_info,
+                _ => unreachable!("chaininfo always respond even if all data inside is `None`"),
+            };
+
+            let expected_difficulty = match chain_info.expected_difficulty {
+                Some(difficulty) => difficulty,
+                _ => Err(Error {
+                    code: ErrorCode::ServerError(0),
+                    message: "Not enough state in the chain".to_string(),
+                    data: None,
+                })?,
+            };
+
             Ok(GetBlockTemplate {
-                capabilities: vec![],
+                capabilities: vec!["proposal".to_string()],
```

Comment:
Optional: set default values in a constructor fn for `GetBlockTemplate`

Reply:
This default value is incorrect, we don't support this feature.